### PR TITLE
Add 'const' keyword in various places. Adopted from Tcl

### PR DESCRIPTION
**Pull request by nijtmans**
Please review

### CODE REVIEW

File: tommath.h [hunk 1]
```diff
@@ -455,9 +455,9 @@ int mp_exptmod(mp_int *a, mp_int *b, mp_int *c, mp_int *d);
 
 /* number of primes */
 #ifdef MP_8BIT
-   #define PRIME_SIZE      31
+#  define PRIME_SIZE      31
```
Comment (sjaeckel) [inline tommath.h]: your PR title is lying :P

Comment (nijtmans) [reply tommath.h]: I just wanted to check your review skills .....   ;-)

**Comment by sjaeckel** [comment]
Probably a stupid question, but you also probably had a good reason: Why does this only take a small subset of the API into account?

**Comment by nijtmans** [comment]
Well, I only did a subset, because Tcl only needed it for a few functions (actually, I did more than just those few). By now, I already did many more (see: https://core.tcl.tk/tcl/info/ae8992c39570e9dd), it just depends on my free time ;-)

**Comment by sjaeckel** [comment]
Alright, my guess was right :-)

Well how about investing some time and doing it overall? How much time do you think it'll take, as you already did it partially!?

**Comment by nijtmans** [comment]
I'm willing to spend some more time on it. I started with the easy ones

**Comment by nijtmans** [comment]
BTW: handling mp_sub_d() is not trivial, due to line 36 and 38:
     a->sign = MP_ZPOS;
     res     = mp_add_d(a, b, c);
     a->sign = c->sign = MP_NEG;
changing the sign on an input value, which is supposed to be const, is a dead-sin ;-)

**Comment by sjaeckel** [comment]
> BTW: handling mp_sub_d() is not trivial, due to line 36 and 38:

well I'd go for something like

```
mp_int a_ = *a;
a_.sign ....
```

**Comment by sjaeckel** [comment]
would you mind closing this PR and creating a fresh one based on this repo and not your private fork?
then I could edit stuff as well easily

**Comment by fperrad** [comment]
my linter find these occurences:
```
bn_fast_s_mp_mul_digs.c  102  Info 818: Pointer parameter 'a' (line 34) could be declared as pointing to const
bn_fast_s_mp_mul_digs.c  102  Info 818: Pointer parameter 'b' (line 34) could be declared as pointing to const
bn_fast_s_mp_mul_high_digs.c  93  Info 818: Pointer parameter 'a' (line 27) could be declared as pointing to const
bn_fast_s_mp_mul_high_digs.c  93  Info 818: Pointer parameter 'b' (line 27) could be declared as pointing to const
bn_fast_s_mp_sqr.c  109  Info 818: Pointer parameter 'a' (line 28) could be declared as pointing to const
bn_mp_cmp_d.c  39  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_cmp_mag.c  50  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_cmp_mag.c  50  Info 818: Pointer parameter 'b' (line 19) could be declared as pointing to const
bn_mp_cnt_lsb.c  47  Info 818: Pointer parameter 'a' (line 23) could be declared as pointing to const
bn_mp_copy.c  62  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_count_bits.c  39  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_div_2.c  63  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_div_3.c  72  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_dr_is_modulus.c  37  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_dr_setup.c  25  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_get_int.c  40  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_get_long.c  40  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_get_long_long.c  40  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_karatsuba_mul.c  169  Info 818: Pointer parameter 'a' (line 47) could be declared as pointing to const
bn_mp_karatsuba_mul.c  169  Info 818: Pointer parameter 'b' (line 47) could be declared as pointing to const
bn_mp_karatsuba_sqr.c  122  Info 818: Pointer parameter 'a' (line 25) could be declared as pointing to const
bn_mp_montgomery_setup.c  53  Info 818: Pointer parameter 'n' (line 19) could be declared as pointing to const
bn_mp_mul_2.c  77  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_mul_d.c  73  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_mp_reduce_is_2k_l.c  38  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_s_mp_sqr.c  79  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_s_mp_sub.c  82  Info 818: Pointer parameter 'a' (line 19) could be declared as pointing to const
bn_s_mp_sub.c  82  Info 818: Pointer parameter 'b' (line 19) could be declared as pointing to const
```

**Pull request closed without merging**